src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,354 @@
+  const contentNavStyles = {
+    display: isHidden ? 'none' : 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    gap: '8px',
+    order: REGION_ORDER.ContextArea
+  }
+  return (
+    <Box hidden={isHidden} sx={merge<BetterSystemStyleObject>(contentNavStyles, sx)}>
+      {children}
+    </Box>
+  )
+}
+
+export type ParentLinkProps = {
+  href?: string
+} & PageHeaderProps
+
+// TODO: add `as` and `aria-label` props
+const ParentLink: React.FC<React.PropsWithChildren<ParentLinkProps>> = ({
+  children,
+  sx = {},
+  href,
+  hidden = onlyVisibleOnNarrowView
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <>
+      <Link
+        muted
+        sx={merge<BetterSystemStyleObject>({display: isHidden ? ' none' : 'flex', alignItems: 'center'}, sx)}
+        href={href}
+      >
+        <ArrowLeftIcon />
+        <Box sx={{marginLeft: 2}}>{children}</Box>
+      </Link>
+    </>
+  )
+}
+
+// ContextBar
+// Generic slot for any component above the title region. Use it for custom breadcrumbs and other navigation elements instead of ParentLink.
+// ---------------------------------------------------------------------
+
+const ContextBar: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = onlyVisibleOnNarrowView
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box hidden={isHidden} sx={sx}>
+      {children}
+    </Box>
+  )
+}
+
+// ContextAreaActions
+// ---------------------------------------------------------------------
+const ContextAreaActions: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = onlyVisibleOnNarrowView
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  console.log({hidden}, {isHidden})
+  return (
+    <Box
+      hidden={isHidden}
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: '8px',
+          flexGrow: '1',
+          justifyContent: 'right'
+        },
+        sx
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// PageHeader.TitleArea: The main title area of the page. Visible on all viewports.
+// PageHeader.TitleArea Sub Components: PageHeader.BackButton, PageHeader.LeadingVisual, PageHeader.Title, PageTitle.TrailingVisual, PageHeader.TrailingActions, PageHeader.Actions
+// PageHeader.BackButton and PageHeader.TrailingAction are only visible on regular viewports therefore they come as hidden on narrow viewports and their visibility can be managed by their exposed `hidden` prop
+// ---------------------------------------------------------------------
+const TitleArea: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}}) => {
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>({display: 'flex', flexDirection: 'row', alignItems: 'center', gap: '8px'}, sx)}
```

Comment:
You mentioned the actions should be aligned to top when title is multiline where cases like this. (Attaching the screenshot you shared)

![Screen Shot 2022-11-22 at 18 12 38](https://user-images.githubusercontent.com/1446503/203876889-bbf09ca9-715a-4cee-bae9-7b0574db56d8.png)

Where do you think the rest of the items align for cases like this? For example if they have `LeadingVisual` and/or `TrailingVisual`?

Reply:
@mperrotti could I please get your thoughts on that? 🙏🏼

Reply:
We'd have to write some custom JS to detect if the lines wrap or not. I tried `align-items: baseline`, and it seems to work well for single line and multi-line titles.

https://user-images.githubusercontent.com/2313998/206012763-9bea8126-950c-48ab-a902-c6e5bcd777d2.mp4

Reply:
The only problem is that the `LeadingVisual` looks a little too high up with `align-items: baseline` on a single line.

I think we need a little more time figuring out which layout styles will give us the best layout for all cases. Maybe we can try playing with the line-height or the top and bottom padding of the icon container?


https://user-images.githubusercontent.com/2313998/206014747-3711f5d6-937d-4789-9447-53be17a9db51.mp4

Reply:
Ah thanks for the exploration, it is great and super helpful! Let me plan with it today and see what I'll come up with!

Reply:
I played with it and giving height to each sub-component's container `height:2rem; alignItems: 'center'` along with giving `alignItems: 'flex-start'` to the main TitleArea container seems to be looking as expected. Here are a few screenshots to show the alignment

## Single Line
<img width="823" alt="Screen Shot 2022-12-07 at 1 46 35 pm" src="https://user-images.githubusercontent.com/1446503/206084239-feb9c9cf-8159-4aa3-acd0-6dbd405ce3f2.png">

## Multi Line
<img width="823" alt="Screen Shot 2022-12-07 at 1 46 15 pm" src="https://user-images.githubusercontent.com/1446503/206084237-a9cad5ef-b793-421f-873b-ffc1c653e122.png">

## When increased the browser's default font-size
<img width="821" alt="Screen Shot 2022-12-07 at 1 46 51 pm" src="https://user-images.githubusercontent.com/1446503/206084242-697b3433-6d80-4e64-9f34-d2e12783cb6c.png">

and here is the commit for these changes : https://github.com/primer/react/pull/2265/commits/98bdc6acd7d91fb500bdddab2814a6daf7131203

What do you think @mperrotti ? 👀

Reply:
This direction seems good to me. Did you choose `2rem` based on the title font size and line height? If so, I'd store `2rem` in a variable and use that variable for setting the heights. If there was another reason you chose `2rem`, what was it?

Also - we'll need to change `2rem` to match the font size. See what happens when I set the title variant to large:
![Screenshot 2022-12-07 at 7 34 52 PM](https://user-images.githubusercontent.com/2313998/206327330-6403beac-b63b-474b-bd9d-a410fdf1a6c0.png)